spec.html

```diff
@@ -2303,6 +2303,17 @@ <h1>Object Internal Methods and Internal Slots</h1>
         <p>Internal method names are polymorphic. This means that different object values may perform different algorithms when a common internal method name is invoked upon them. That actual object upon which an internal method is invoked is the &ldquo;target&rdquo; of the invocation. If, at runtime, the implementation of an algorithm attempts to use an internal method of an object that the object does not support, a *TypeError* exception is thrown.</p>
         <p>Internal slots correspond to internal state that is associated with objects and used by various ECMAScript specification algorithms. Internal slots are not object properties and they are not inherited. Depending upon the specific internal slot specification, such state may consist of values of any ECMAScript language type or of specific ECMAScript specification type values. Unless explicitly specified otherwise, internal slots are allocated as part of the process of creating an object and may not be dynamically added to an object. Unless specified otherwise, the initial value of an internal slot is the value *undefined*. Various algorithms within this specification create objects that have internal slots. However, the ECMAScript language provides no direct way to associate internal slots with an object.</p>
+        <p>All objects have an internal slot named [[PrivateFields]], which is a List of PrivateField Records. This List represents the values of the private fields for the object. Initially, it is an empty List.</p>
+        <p>All objects have an internal slot named [[PrivateBrands]], which is a List of ECMAScript values which are used for checking to see if an object supports a private method or accessor. The entries in the list correspond to the "original prototype" (or, in the case of static methods, the "original constructor") which contained the private method or accessor. Initially, it is an empty List.</p>
+        <emu-note>
+          <p>Although this specification uses logic of a [[PrivateBrands]] List which is held per-instance, the use of this list obeys certain invariants:</p>
+
+          <ul>
+            <li>Nothing is removed from the list; things are only added to the end.</li>
+            <li>There is no way to distinguish different copies of the list.</li>
+            <li>The list is mutable, but there are never any references to it outside of the particular object which holds it, so additions can be thought of as replacing the internal slot with another list which contains the new element.</li>
```

Comment:
It's going to be really hard to preserve this invariant. Can we just say something more along the lines of "though this list is mutated, it is always used in such a way that updates to the list are equivalent to replacing the value of the slot with a new list"?

Reply:
Obviated by https://github.com/tc39/ecma262/pull/1668/commits/558a63cbec0e6d0de22091664de3f17e0e4eccd7.